e2e/src/common.rs

```diff
@@ -33,60 +28,6 @@ pub trait FutureWaitExt: Future + Sized {
 
 impl<F> FutureWaitExt for F where F: Future {}
 
-pub trait FutureBuilderExt: Sized {
-    type Future: Future;
```

Comment:
We can remove the `type Future: Future` and used `BoxFuture` instead. I'm not sure I understand why this was removed altogether.

Reply:
To get this to work I had to change for example

```rust
impl<T, R> FutureBuilderExt for MethodBuilder<T, R>
where
    T: Transport + 'static + Send + Sync,
    <T as Transport>::Out: Send,
    R: Detokenizable + 'static + Send,
{
    type Future = BoxFuture<'static, Result<TransactionResult, MethodError>>;

    fn into_future(self) -> Self::Future {
        self.send().boxed()
    }
}
```

It felt like that wasn't worth it just so we can call `wait` directly instead of `send().wait`.

Reply:
I suggest using `.local_boxed()` instead so you don't need the `Send` bounds, which shouldn't be needed here in any case.